### PR TITLE
The tgt device update function handles bsopts

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn/longhorn#1674

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

File: usr/spc.c [hunk 1]
```diff
@@ -1972,6 +1972,47 @@ tgtadm_err spc_lu_offline(struct scsi_lu *lu)
...
+	if (end == NULL)
+		end = *p + strlen(*p);
+	len = end - *p;
+	ret = malloc(len + 1);
+	strncpy(ret, *p, len);
+	ret[len] = '\0';
+	*p = end;
+	/* Jump past the semicolon, if we stopped at one */
+	if (**p == ';')
+		*p = end + 1;
```
Comment (PhanLe1010) [inline usr/spc.c:1991]: Could you please explain why we need to modify this pointer? 

I don't see that we reuse that pointer for the next token as we overwrite the buf here https://github.com/rancher/tgt/pull/11/files#diff-5280d48ef04f28efc512890b97c623410c797e3d78bc682a7d43e332c6e1b1a0R2118

Comment (shuo-wu) [reply usr/spc.c:1991]: These functions are copied from other files hence I didn't check this detail... Actually there are multiple vendor implementations containing them. I don't know why no one refactors them as util functions in a separate file.

Comment (PhanLe1010) [reply usr/spc.c:1991]: I see. I see it doesn't hurt but was a bit curious. Thank you

**Pull request merged**